### PR TITLE
Generate Comparison Charts 

### PR DESCRIPTION
**Pull request by MVarshini**
PBENCH-1208

Generate Comparison Charts

### CODE REVIEW

File: dashboard/src/modules/components/ComparisonComponent/PanelContent.jsx [hunk 1]
```diff
@@ -1,36 +1,82 @@
...
+    (item) => {
+      if (searchValue === "") {
+        return true;
+      }
+      let input;
+      try {
+        input = new RegExp(searchValue, "i");
+      } catch (err) {
+        input = new RegExp(
+          searchValue.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"),
```
Comment (dbutenhof) [inline dashboard/src/modules/components/ComparisonComponent/PanelContent.jsx:27]: So you're allowing RegEx characters in the user's filter string, but if it doesn't compile you're quoting all special characters and compiling again? I'm not sure this is what I'd want.

* If I use regex characters accidentally, quoting them is good, but probably not as a fallback. That is, if I didn't intentionally construct a regex, then the results of `AB*C` will surprise me;
* If I construct a regex but get it wrong, quoting all special characters will probably still surprise me.

I think that if we're not going to allow regex input, then you should quote (or reject) special characters up front (or simply use a non-regex filter), and if we intentionally allow regex then a compile failure should be an error visible to the user so they can retry.

I also noticed that in the demos you were using this to find the `uperf` benchmark runs. As mentioned earlier, using the name is unreliable here: it'd be better to search for an exact match in `server.benchmark`... although I can imagine users finding name search useful as well, I suspect that in general they'll need a lot more.

I don't think we want to get into this for this PR, but something to keep in mind is that I think perhaps in the longer run what we want is a general search component for datasets that you can use on each page that allows filtering by wildcard (regex?) name match or by arbitrary metadata. That is, when I get to the comparison I probably don't want to see all my/public `uperf` but rather the set of `uperf` data I generated for a specific purpose, which might be based on my config string or a `global.lightspeed.run` value or something.

**Review by dbutenhof — CHANGES_REQUESTED** [review]
This mostly looks OK, but I think the handling of the search box (e.g., implicitly accepting regex but quietly quoting all the special characters on a compilation failure) needs some clarification. I've got a few other minor comments, too, including a debugging `console.log` which should probably be removed. 😁

**Review by webbnh — APPROVED** [review]
I didn't find anything worth blocking the merge for, but I've pointed out a few code quality issues that you might want to consider addressing.

**Review by dbutenhof — APPROVED** [review]
I would prefer the compare pane search be cleaned up, but let's move this forward so we can stage it soon.

**Pull request merged**